libtenzir/builtins/functions/duration.cpp

```diff
@@ -75,8 +75,183 @@ class plugin final : public function_plugin {
     });
   }
 };
+
+class into_duration_plugin final : public function_plugin {
+public:
+  into_duration_plugin() = default;
+
+  into_duration_plugin(std::string name, tenzir::duration unit)
+    : name_{std::move(name)}, unit_{unit} {
+  }
+
+  auto name() const -> std::string override {
+    return name_;
+  }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    TRY(argument_parser2::function(name())
+          .positional("x", expr, "duration")
+          .parse(inv, ctx));
+    return function_use::make([this, expr = std::move(expr)](
+                                evaluator eval, session ctx) -> series {
+      auto b = duration_type::make_arrow_builder(arrow::default_memory_pool());
+      check(b->Reserve(eval.length()));
+      for (auto& arg : eval(expr)) {
+        match(
+          *arg.array,
+          [&](const arrow::NullArray& arg) {
+            check(b->AppendNulls(arg.length()));
+          },
+          [&](const arrow::DoubleArray& arg) {
+            constexpr auto min
+              = static_cast<double>(
+                  std::numeric_limits<tenzir::duration::rep>::lowest())
+                - 1.0;
+            constexpr auto max
+              = static_cast<double>(
+                  std::numeric_limits<tenzir::duration::rep>::max())
+                + 1.0;
```

Comment:
How can this protect against overflow if `unit_` is not taken into account?

Reply:
Isn't it checked on line 129?

Reply:
Seems like there previously were two checks, now there's only one. The remaining one looks better :+1: